models/helper.go

```diff
@@ -82,3 +82,20 @@ func QueryFilterByCohortPairsHelper(filterCohortPairs []utils.CustomDichotomousV
 	query := resultsDataSource.Db.Table(unionAndIntersectSQL+" as "+unionAndIntersectSQLAlias+" ", idsList...)
 	return query
 }
+
+// This function will get the concept information for given conceptId, and
+// return the best SQL to use for doing a "not null" check on its value in the
+// observation table.
+func GetConceptValueNotNullCheckBasedOnConceptType(observationTableAlias string, sourceId int, conceptId int64) string {
+	conceptModel := *new(Concept)
+	conceptInfo, error := conceptModel.RetrieveInfoBySourceIdAndConceptId(sourceId, conceptId)
+	if error != nil {
+		panic("error while trying to get information for conceptId, or conceptId not found")
+	} else if conceptInfo.ConceptType == "MVP Continuous" {
+		return observationTableAlias + ".value_as_number is not null"
+	} else if conceptInfo.ConceptType == "MVP Ordinal" {
+		return observationTableAlias + ".value_as_concept_id is not null and " + observationTableAlias + ".value_as_concept_id != 0"
+	} else {
+		return observationTableAlias + ".value_as_string is not null"
```

Comment:
I'd make a note about this because i'm not sure if this is true; but we aren't using those types yet anyways; just that we will want to verify this once we do

Reply:
good catch. We can also just raise an error in this case.... maybe that's better for now.

Reply:
done. The code raises an error now in this situation.